rust/connlib/snownet/src/node.rs

```diff
@@ -1406,11 +1470,12 @@ impl Connection {
                 self.stats.stun_bytes_to_peer_direct += packet.len();
 
                 // `source` did not match any of our allocated sockets, must be a local one then!
-                return Some(Transmit {
+                transmits.push_back(Transmit {
                     src: Some(source),
                     dst,
                     payload: Cow::Owned(packet.into()),
                 });
```

Comment:
So instead of returning and being polled again multiple times, now all these `Transmit` events are pushed into a VecDeque, right?

Any chance the VecDeque can overflow or anything? Like, use an unlimited amount of memory if something is scheduled in a strange order?

Reply:
There is no defense in `snownet` against it overflowing but `Tunnel` should empty this completely before calling any other APIs which means it should be able to overflow.